### PR TITLE
CLOUDSTACK-8443: Support CentOS 7 as KVM hypervisor

### PR DESCRIPTION
**Pull request by remibergsma**
This adds support for CentOS 7 as a KVM hypervisor. As discussed in the Jira issue, the root cause of the issue was the `cloudstack-setup-agent` script that added this line to `/etc/libvirt/qemu.conf`:

`cgroup_controllers=["cpu"]`

On CentOS 6 this 'works', but since by default no cgroups are mounted, this setting is ignored by `libvirtd`. On CentOS 7 is does not work, because the `cpu` and `cpuacct` are 'co-mounted'. This simpy means you cannot use one without the order. The config line above instructs `libvirtd` to only use the `cpu` cgroup, which it can't.

There was code added to make sure the `cpu` cgroup was not co-mounted. It did this with a shell script, but after a reboot it was gone. It wasn't reliable and not even a solution at all. I went ahead and removed the code, because the co-mounted cgroups work fine. The real issue was the config line we put in `/etc/libvirt/qemu.conf` so much better to fix that instead.

The default for `libvirtd` is to use only cgroups that are mounted. When you remove the 'cgroup_controllers' line, the default kicks in. According to the `qemu.conf` file, the default is:

`cgroup_controllers = ["cpu", "devices", "memory", "blkio", "cpuacct", "net_cls"]`

Then again, if it's not mounted, it's not used. So, that's a nice way to control it.
This works just fine for both RHEL/CentOS 6 and RHEL/CentOS 7.

Along the debugging, I came across some small issues here and there which I also fixed in this PR.

If you test this, be sure to build RPMs from this branch and install those on the KVM hypervisor.

At Schuberg Philis we run KVM hypervisors on CentOS 7 for months with these fixes. Now I took the time to fix the issues in CloudStack, rather than work around them.

@bhaisaab @karuturi could you please review this? Thanks!

Please note:
The next step is to fix CLOUDSTACK-8625 (Systemd profile for KVM Agent) as currently a sysvinit script is still used. Although it works, we do not have any systemd benefits. As it is a separate issue, I'll try to address it in a separate PR.

### CODE REVIEW

**Comment by borisroman** [comment]
LGTM -> Deployed on Centos7 started 2 systemvms, 1 router and 1 guest vm without libvirt errors.

**Comment by bhaisaab** [comment]
@remibergsma looks good, but let me test it and get back to you on this; also for a small patch a single commit is preferred as multiple commits dilute the main motive of the fix when someone is browsing git history or referencing in future - 'Supporting CentOS7 as KVM hypervisor host'

**Comment by remibergsma** [comment]
@borisroman @bhaisaab Thanks for testing! 

I'll have another look at the commits and see if it makes sense to quash some of them together (I'll make them atomic).

**Comment by bhaisaab** [comment]
just to clarify I've not tested this against a real env, just reviewed the code; there are few cgroup related things I don't completely grok but I'll invest time on them

**Comment by remibergsma** [comment]
Squashed some commits that belong to each other.

**Comment by remibergsma** [comment]
@bhaisaab No worries, took me some time as well to understand how it is supposed to work and I'm also still learning.

In the Jira issue you mention two issues. The first:
```
org.libvirt.LibvirtException: Controller 'cpuacct' is not wanted, but 'cpu' is co-mounted: Invalid argument
```

This is due to the co-mount: both 'cpu' and 'cpuacct' are on one mount point, and our previous config instructs it to use only 'cpu'. So, the 'cpuacct' is `not wanted` by our config. This was solved in my PR by relying on the default.

The second issue:
```
unsupported configuration: CPU tuning is not available on this host
```

This issue also went away by updating the `libvirtd` config. When I look at the Libvirt source (https://github.com/virtualopensystems/libvirt/blob/master/src/qemu/qemu_cgroup.c#L635) I see this error is thrown in qemu_cgroup.c I assume the VIR_ERR_CONFIG_UNSUPPORTED aka `unsupported configuration` also has to do with the rather weird cgroup config that we put there before but I don't know for sure.

If you have some time, please give it a go. I'll also ask @wilderrodrigues to review it.

**Comment by karuturi** [comment]
This is what I have done to test

installed a fresh setup with centos63 rpms on MS with 6.3 and KVM host with 6.3. 
verified that systemvms, VR and user instances successfully started.
verified that cgroup_controllers=["cpu"] doesnt exists at host:/etc/libvirt/qemu.conf

installed a fresh setup with centos7 rpms on MS with 7 and KVM host with 7.
verified that systemvms, VR and user instances successfully started.
verified that cgroup_controllers=["cpu"] doesnt exists at host:/etc/libvirt/qemu.conf

:+1: 

Thanks for the detailed description :)

**Comment by remibergsma** [comment]
Thanks for testing and verifying @karuturi :-)

File: plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java [hunk 1]
```diff
@@ -699,17 +698,6 @@ public boolean configure(final String name, final Map<String, Object> params) th
             _hypervisorType = HypervisorType.KVM;
         }
 
-        //Verify that cpu,cpuacct cgroups are not co-mounted
-        if(HypervisorType.LXC.equals(getHypervisorType())){
```
Comment (bhaisaab) [inline plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java]: @kishankavala can you review this patch, it removes some of the cgroup setting stuff for LXC?

Comment (remibergsma) [inline plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java]: @bhaisaab I just tested LXC and it works just fine with the co-mounted cgroups. The problem this code works-around has been solved. I think we're fine.

**Comment by kishankavala** [comment]
@remibergsma changes looks fine. I would like to test it and get back to you, hopefully by tomorrow.

**Comment by remibergsma** [comment]
@kishankavala Thanks, that would be great! Let me know if I can be of any help.

**Pull request merged**